tests/integration/cluster_test.go

```diff
@@ -518,3 +518,51 @@ func TestSpeedyTerminate(t *testing.T) {
 	case <-donec:
 	}
 }
+
+// TestConcurrentRemoveMember demonstrated a panic in mayRemoveMember with
+// concurrent calls to MemberRemove. To reliably reproduce the panic, a delay
```

Comment:
There should never be a concurrent call to `MemberRemove`. Cluster state changes should be all done via apply loop, which guarantees that there is only one concurrent change to cluster state. 

The reason for locking in cluster struct (and any other internal structs) is to prevent reader+writer races. There should never be writer+writer race.

Reply:
This PR has nothing to do with the apply loop/workflow. It's fixing a panic in the API layer. Refer to https://github.com/etcd-io/etcd/pull/18606#issuecomment-2367970699

Reply:
Oh, I didn't notice that this is integration test. I'm just little confused why MemberRemove call by client is executed concurrently. Should it be linearized?

Reply:
> I'm just little confused why MemberRemove call by client is executed concurrently.

Right, usually we don't expect users to do this. But etcdserver shouldn't panic due to whatever users' inappropriate behaviour. Note that the panicking happens at API layer instead of the apply loop/workflow.

Reply:
Sorry, didn't mean "why?", but "how?". Agree that this is a problem, but I'm trying to understand how but in cluster struct surfaces to API.